### PR TITLE
feat: enable CLI tests

### PR DESCRIPTION
**Pull request by christiancp100**
### What does it do?
It adds the first CLI tests for Data Transfer.

### Why is it needed?

To test data transfer CLi.

### How to test it?

Execute `yarn test:cli` at the root level of the process.

### CODE REVIEW

**Review by Convly — COMMENTED** [review]
On my side, trying to run the test fails [with the following errors](https://github.com/strapi/strapi/assets/25851739/ee7e3159-dc5c-47b0-a14a-48e4d4e0d810)

**Comment by christiancp100** [comment]
> Member

Yeah that's what I commented in the Slack team channel, It's the same error we have when running the Api tests locally. In order to make them work locally you have to create a `yarn.lock` file and do `yarn install` inside the test-app-0. I couldn't find out why that's happening, but in the CI it seems to work for the API Tests

**Review by innerdvations — CHANGES_REQUESTED** [review]
LGTM! The only thing we need is a detailed comment about why we have to use 'npm' here instead of 'yarn' so that we don't have to rip our hair out in the future, and I'll click 'approve' :laughing:

**Review by Convly — CHANGES_REQUESTED** [review]
Seems like the tests are failing but are not reported as such by the task

image

**Comment by christiancp100** [comment]
> LGTM! The only thing we need is a detailed comment about why we have to use 'npm' here instead of 'yarn' so that we don't have to rip our hair out in the future, and I'll click 'approve' 😆

What would you say is the most suitable place to put this?

File: cli-tests/tests/strapi/data-transfer/export.test.cli.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  let appPath;
+  const outputFilename = 'output';
+
+  beforeAll(async () => {
+    const testApps = utils.instances.getTestApps();
+
+    appPath = testApps.at(0);
+  });
+
+  it('should export the data', async () => {
```
Comment (innerdvations) [inline cli-tests/tests/strapi/data-transfer/export.test.cli.js:17]:
```suggestion
  it('should export data without error', async () => {
```

File: cli-tests/tests/strapi/data-transfer/import.test.cli.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          outputFilename,
+        ],
+        {
+          cwd: appPath,
+        }
+      )
+      .end();
+  });
+
+  it('should import the data', async () => {
```
Comment (innerdvations) [inline cli-tests/tests/strapi/data-transfer/import.test.cli.js:36]:
```suggestion
  it('should prompt for confirmation before importing data', async () => {
```

File: cli-tests/tests/strapi/data-transfer/import.test.cli.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          stdio: 'inherit',
+        }
+      )
+      .waitForPrompt()
+      .write('Y\n')
+      .expect('code', 0)
+      .end();
+  });
+
+  it('should import the data without asking for confirmation', async () => {
```
Comment (innerdvations) [inline cli-tests/tests/strapi/data-transfer/import.test.cli.js:52]:
```suggestion
  it('should import data without asking for confirmation with --force', async () => {
```

File: cli-tests/tests/strapi/version/version.test.cli.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  let currentVersion;
+
+  beforeAll(async () => {
+    const testApps = utils.instances.getTestApps();
+
+    appPath = testApps.at(0);
+    currentVersion = require(`${appPath}/package.json`).dependencies['@strapi/strapi'];
+  });
+
+  it('should output version', async () => {
```
Comment (innerdvations) [inline cli-tests/tests/strapi/version/version.test.cli.js:18]:
```suggestion
  it('should output version with argument', async () => {
```

File: test/scripts/run-cli-tests.js [hunk 5]
```diff
@@ -208,32 +212,43 @@ yargs
...
+                await execa(
+                  'jest',
+                  [
+                    '--config',
+                    '../../../jest.config.cli.js',
+                    '--rootDir',
+                    domainDir,
+                    '--color',
+                    '--verbose',
+                    '--runInBand',
```
Comment (innerdvations) [inline test/scripts/run-cli-tests.js:230]:
```suggestion
                    '--runInBand', // tests must not run concurrently
```

**Review by innerdvations — COMMENTED** [review]
Awesome! Just a few text comments

**Review by Convly — APPROVED** [review]

**Pull request merged**